File: ProjectEuler/Problem_3/3-Largest_prime_factor.cs
void Main()
{
	double x = 600851475143;
	for (int i = (int) Math.Sqrt(x)+1; i > 3; i -= 2) {
		if (x % i == 0 && isPrime(i)) {
			Console.WriteLine("Largest prime factor of {0} is {1}", x, i);
			break;
		}
	}
}

bool isPrime(double num)
{
	if (num < 2) return false;
	if (num == 2 || num == 3) return true;
	if (num % 2 == 0 || num % 3 == 0) return false;
	foreach (int i in Enumerable.Range(3, (int) Math.Sqrt(num) + 1))
	{
		if (num % i == 0) return false;
	}
	return true;
}
// Largest prime factor of 600851475143 is 6857

File: ProjectEuler/Problem_3/3-Largest_prime_factor.js
// What is the largest prime factor of the number 600851475143 ?
 
const check_prime = x => {
    for (let i = Math.floor(Math.sqrt(x) + 1); i > 3; i -= 2){
        if (x % i == 0 && is_prime(i))
            return i    
    }
}

const is_prime = x => {
    if (x < 2) return false
    if (x == 2 || x == 3 || x == 5) return true
    if (x % 2 == 0 || x % 3 == 0 || x % 5 == 0) return false
    for (let i = 3; i < Math.sqrt(x) + 1; i += 2){
        if (x % i == 0) return false
    }
    return true
}
console.log(is_prime(13))
console.log(check_prime(600851475143))